docs/api.rst

```diff
@@ -406,7 +406,25 @@ Array form of :ref:`zip`.  Apply a function to corresponding events from all the
   s2:                           --1--2--3->
   zipArray(add3, [s1, s2, s3]): --3--6--9->
 
-sample :: ((a, b) -> c) -> Stream a -> Stream b -> Stream c
+sample
+^^^^^^
+.. code-block:: haskell 
+
+  sample :: ((a, b) -> c) -> Stream a -> Stream b -> Stream c
+
+Create a new stream by combining sampled values from many input streams.::
```

Comment:
@Frikki That's a great point.  I agree: trying to describe in 1 sentence is a good goal, but we shouldn't force it if 2-3 sentences makes for a better explanation.

On that note, how about something like:

> For each event in a _sampler_ stream, apply a function to combine it with the most recent event in another stream.  The resulting stream will contain the same number of events as the _sampler_ stream.

I also thought of this, which may be a more compact way to say the first sentence above:

> Combine each event in one stream with the most recent event in another.

Maybe we could also point out the difference between `sample` and `combine`.

Do either of those seem like a good direction?  Do they spark other ideas?